src/Storages/StorageURL.cpp

```diff
@@ -451,10 +477,23 @@ StorageURLSink::StorageURLSink(
     std::string content_type = FormatFactory::instance().getContentType(format, context, format_settings);
     std::string content_encoding = toContentEncodingName(compression_method);
 
+    auto proxy_config = getProxyConfiguration(http_method);
+
+    auto write_buffer = WriteBufferFromHTTPBuilder()
+        .withURI(Poco::URI(uri))
+        .withMethod(http_method)
+        .withContentType(content_type)
+        .withContentEncoding(content_encoding)
+        .withAdditionalHeaders(headers)
+        .withTimeouts(timeouts)
+        .withProxyConfiguration(proxy_config)
+        .build();
+
     write_buf = wrapWriteBufferWithCompressionMethod(
-        std::make_unique<WriteBufferFromHTTP>(Poco::URI(uri), http_method, content_type, content_encoding, headers, timeouts),
+        std::move(write_buffer),
```

Comment:
IMO it is much better to avoid using `WriteBufferFromHTTPBuilder` and just do:
```suggestion
    auto write_buffer = std::make_unique<WriteBufferFromHTTP>(Poco::URI(uri), http_method, content_type, content_encoding, headers, timeouts, DBMS_DEFAULT_BUFFER_SIZE, proxy_config),)
```

Sadly, we can't use designated initializers with `std::make_unique`, but let's not bring builders just for a single-use convenience.